clang-tools-extra/clangd/ClangdServer.cpp

```diff
@@ -624,6 +624,50 @@ void ClangdServer::rename(PathRef File, Position Pos, llvm::StringRef NewName,
   WorkScheduler->runWithAST("Rename", File, std::move(Action));
 }
 
+void ClangdServer::indexedRename(
+    std::map<URIForFile, std::vector<Position>> Positions, PathRef PrimaryFile,
+    llvm::StringRef OldName, llvm::StringRef NewName, Callback<FileEdits> CB) {
+  ParseInputs Inputs;
+  Inputs.TFS = &TFS;
+  Inputs.CompileCommand = CDB.getCompileCommand(PrimaryFile)
+                              .value_or(CDB.getFallbackCommand(PrimaryFile));
+  IgnoreDiagnostics IgnoreDiags;
+  std::unique_ptr<CompilerInvocation> CI =
+      buildCompilerInvocation(Inputs, IgnoreDiags);
+  if (!CI) {
+    return CB(llvm::make_error<llvm::StringError>(
+        "Unable to get compiler arguments for primary file",
+        llvm::inconvertibleErrorCode()));
```

Comment:
We're only using them to determine if ObjC or not, is it worth falling back to just checking for `:`?

Reply:
We also use it below in `editsForLocations`, which uses the `LangOpts` to tokenize the file.